pkg/controller/reconciler/flux_controller.go

```diff
@@ -0,0 +1,431 @@
+	_, err := r.Bicep.BuildParams(bicepParamsFile, "--outfile", outfile)
+	if err != nil {
+		logger.Error(err, "failed to run bicep build-params")
+		return map[string]any{}, err
+	}
+
+	// Read the contents of the generated .parameters.json file
+	contents, err := r.FileSystem.ReadFile(outfile)
+	if err != nil {
+		logger.Error(err, "failed to read bicep build-params output")
+		return nil, err
+	}
+
+	params := make(map[string]any)
+	err = json.Unmarshal(contents, &params)
+	if err != nil {
+		logger.Error(err, "failed to unmarshal bicep build-params output")
+		return nil, err
+	}
+
+	if params[armJSONParametersKeyName] == nil {
+		return nil, fmt.Errorf("parameters field not found in bicep build-params output")
+	}
+
+	if _, ok := params[armJSONParametersKeyName].(map[string]any); !ok {
+		typeGot := fmt.Sprintf("%T", params[armJSONParametersKeyName])
+		return nil, fmt.Errorf("unexpected format for parameters field in bicep build-params output, got %s", typeGot)
+	}
+
+	parameters := params[armJSONParametersKeyName].(map[string]any)
+
+	return parameters, nil
+}
+
+// createOrUpdateDeploymentTemplate creates or updates a DeploymentTemplate object in the cluster
+// with the given spec.
+func (r *FluxController) createOrUpdateDeploymentTemplate(ctx context.Context, fileName, namespace, template, providerConfig string, parameters map[string]string, repository string) error {
+	logger := ucplog.FromContextOrDiscard(ctx)
+
+	// Try to get the DeploymentTemplate object from the cluster
+	deploymentTemplate := radappiov1alpha3.DeploymentTemplate{}
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: fileName}, &deploymentTemplate)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			// Error getting the DeploymentTemplate object that is not a NotFound error
+			logger.Error(err, "unable to get deployment template")
+			return err
+		}
+
+		// If the DeploymentTemplate doesn't exist, create it
+		deploymentTemplate := &radappiov1alpha3.DeploymentTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fileName,
+				Namespace: namespace,
+			},
+			Spec: radappiov1alpha3.DeploymentTemplateSpec{
+				Template:       template,
+				Parameters:     parameters,
+				ProviderConfig: providerConfig,
+				Repository:     repository,
+			},
+		}
+		if err := r.Client.Create(ctx, deploymentTemplate); err != nil {
+			logger.Error(err, "unable to create deployment template")
+			return err
+		}
+
+		logger.Info("Created Deployment Template", "name", deploymentTemplate.Name)
+		return nil
+	}
+
+	// If the DeploymentTemplate already exists, update it
+	deploymentTemplate.Spec = radappiov1alpha3.DeploymentTemplateSpec{
+		Template:       template,
+		Parameters:     parameters,
+		ProviderConfig: providerConfig,
+		Repository:     repository,
+	}
+	if err := r.Client.Update(ctx, &deploymentTemplate); err != nil {
+		logger.Error(err, "unable to update deployment template")
```

Comment:
Should there be a `return err` here, or is the intent to continue and log the error?

Reply:
missed this, thanks